spec/models/lettings_log_spec.rb

```diff
@@ -1834,6 +1834,14 @@ def previous_postcode_fields
         expect(lettings_log.beds_for_la_rent_range).to eq(4)
       end
     end
+
+    context "when the log is for a bedsit, beds is not routed to and is not yet re-derived" do
+      let(:lettings_log) { build(:lettings_log, unittype_gn: 2, beds: nil) }
+
+      it "returns 1" do
+        expect(lettings_log.beds_for_la_rent_range).to eq(1)
+      end
+    end
```

Comment:
So I don't think this test is actually testing anything meaningful. If beds is set to nil here, we're not actually testing the code that clears the beds value and I think if we only build and not create the log that also means that we're not running any of the before_validation callbacks, which from what I'm gathering is what causes the issue in the first place (?)